test/server/options_impl_test.cc

```diff
@@ -1,55 +1,41 @@
+#include "common/common/utility.h"
 #include "server/options_impl.h"
 
-TEST(OptionsImplDeathTest, HotRestartVersion) {
+// Do the ugly work of turning a std::string into a char** and create an OptionsImpl. Args are
+// separated by a single space: no fancy quoting or escaping.
+std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+  std::vector<std::string> words = StringUtil::split(args, ' ');
   std::vector<const char*> argv;
-  argv.push_back("envoy");
-  argv.push_back("--hot-restart-version");
-  EXPECT_EXIT(OptionsImpl(argv.size(), const_cast<char**>(&argv[0]), "1", spdlog::level::warn),
-              testing::ExitedWithCode(0), "");
+  for (const std::string& s : words) {
+    argv.push_back(s.c_str());
```

Comment:
Sigh, sorry, I'm not having a good code review day. I need to spend more time looking. Yes this is fine.